### DISCUSSION TITLE
Get public AND private repositories of a user

### DISCUSSION THREAD

**Discussion by philmetzger** [community/community#69591] [Apps, API and Webhooks]
### Select Topic Area

Question

### Body

I am trying to get public and private repos of a user.

So I authorize a user using this user to get the `code`:
`https://github.com/login/oauth/authorize?scope=read:user&client_id=${process.env.NEXT_PUBLIC_GITHUB_APP_CLIENT_ID}&redirect_uri=${redirectUrl}`

I then get the token using the following:
```
const octokit = new Octokit({
    authStrategy: createOAuthAppAuth,
    auth: {
      clientId: process.env.NEXT_PUBLIC_GITHUB_APP_CLIENT_ID,
      clientSecret: process.env.GITHUB_APP_CLIENT_SECRET,
    },
  });
const result = await octokit.auth({
      type: 'oauth-user',
      code,
    });
const { token } = result as any;
```

Then I do the following to get repos of a user:
```
const responseRepos = await octokitUser.request('GET /user/repos', {
      headers: {
        'X-GitHub-Api-Version': API_VERSION,
        authorization: `Bearer ${token}`,
      },
    });
```

However it only returns public repos and not private. 
Anyone know whats wrong?

**Comment by moayad-star** [comment]
It seems like you're trying to use GitHub's OAuth authentication to access a user's repositories. The issue you're facing, where you're only able to retrieve public repositories and not private ones, might be due to the scope you requested during the authorization step.

In your authorization URL, you've specified the scope as `read:user`, which grants access to the user's profile information but not their private repositories. To access a user's private repositories, you need to request the `repo` scope in addition to `read:user`. Modify your authorization URL like this:

```javascript
https://github.com/login/oauth/authorize?scope=read:user,repo&client_id=${process.env.NEXT_PUBLIC_GITHUB_APP_CLIENT_ID}&redirect_uri=${redirectUrl}
```

Make sure that the user authorizes your application with these additional permissions when they log in. Once you have the updated access token, you should be able to retrieve both public and private repositories using the same API request:

```javascript
const responseRepos = await octokitUser.request('GET /user/repos', {
  headers: {
    'X-GitHub-Api-Version': API_VERSION,
    authorization: `Bearer ${token}`,
  },
});
```

With the `repo` scope granted during authorization, you should be able to access both public and private repositories for the authenticated user.

**Reply by philmetzger** [reply]
Hmm but that gives me FULL read AND WRITE access which would be overboard, i just want read access.

**Reply by philmetzger** [reply]
Also when I added that scope with "repo", it didnt make a difference.
I event deleted the oauth connection in github and reconnected but still i dont see private repos.

**Reply by moayad-star** [reply]
I apologize for any confusion. You are correct that the `repo` scope provides both read and write access, which may be excessive if you only need read access. To access a user's private repositories with read-only access, you should use the `repo:status` scope instead of just `repo`. The `repo:status` scope allows read access to private repositories without granting write access.

Here's the modified authorization URL:

```javascript
https://github.com/login/oauth/authorize?scope=read:user,repo:status&client_id=${process.env.NEXT_PUBLIC_GITHUB_APP_CLIENT_ID}&redirect_uri=${redirectUrl}
```

Make sure to request this updated scope during the authorization process, and then obtain an access token. Once you have the updated token, try accessing the user's repositories again:

```javascript
const responseRepos = await octokitUser.request('GET /user/repos', {
  headers: {
    'X-GitHub-Api-Version': API_VERSION,
    authorization: `Bearer ${token}`,
  },
});
```

This should allow you to retrieve both public and private repositories with read-only access. If you're still facing issues with private repositories, ensure that the user has granted the necessary permissions during the authorization process and that your OAuth application has the required access.

If you continue to encounter problems, you may want to double-check the GitHub OAuth application settings and verify that the necessary permissions are correctly configured.

**Reply by philmetzger** [reply]
dude you are just using chatgpt which doesnt help!

**Reply by moayad-star** [reply]
I don't know how to solve it, so I used artificial intelligence, and I think it disappointed me

**Reply by samus-aran** [reply]
Hi @moayad-star, 👋  

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Reply by moayad-star** [reply]
Thanks @samus-aran , I have read the rules and understand the matter. This is a good solution to ensure the safety of the community. GitHub. Thank you again for alerting me to the rules and I will make sure that the mistake does not happen again.

**Comment by philmetzger** [comment]
My solution to this btw was to install the app which then seems to give me access to private repos.

If anyone comes across this and wants more guidance then reply here :)
